nav2_controller/src/controller_server.cpp

```diff
@@ -569,16 +570,19 @@ void ControllerServer::publishVelocity(const geometry_msgs::msg::TwistStamped &
 
 void ControllerServer::publishZeroVelocity()
 {
-  geometry_msgs::msg::TwistStamped velocity;
-  velocity.twist.angular.x = 0;
-  velocity.twist.angular.y = 0;
-  velocity.twist.angular.z = 0;
-  velocity.twist.linear.x = 0;
-  velocity.twist.linear.y = 0;
-  velocity.twist.linear.z = 0;
-  velocity.header.frame_id = costmap_ros_->getBaseFrameID();
-  velocity.header.stamp = now();
-  publishVelocity(velocity);
+  if (get_parameter("publish_zero_velocity").as_bool()) {
```

Comment:
Store as a class member, I don't like getting parameters in the application

Also, please implement analog to the main branch implementation: the `if` check should be where `publishZeroVelocity` is called, not internal to the function.

Reply:
- [x] Add publish_zero_velocity as class member,
- [x] Add IF where the function  publishZeroVelocity is called